src/phoenix/server/api/schema.py

```diff
@@ -282,10 +279,42 @@ def span_evaluation_summary(
         self,
         info: Info[Context, None],
         evaluation_name: str,
+        time_range: Optional[TimeRange] = UNSET,
+        filter_condition: Optional[str] = UNSET,
     ) -> Optional[EvaluationSummary]:
         if (evals := info.context.evals) is None:
             return None
-        evaluations = evals.get_span_evaluations_by_name(evaluation_name)
+        if (traces := info.context.traces) is None:
+            return None
+        predicate = (
+            SpanFilter(
+                condition=filter_condition,
+                evals=evals,
+            )
+            if filter_condition
+            else None
+        )
+        span_ids = evals.get_span_evaluation_span_ids(evaluation_name)
+        if not span_ids:
+            return None
+        spans = traces.get_spans(
+            start_time=time_range.start if time_range else None,
+            stop_time=time_range.end if time_range else None,
+            span_ids=span_ids,
```

Comment:
might be cleaner if this is written more functional rather than have the method do the work

```
traces.get_spans_by_id(ids).filter(timerange)
```

Reply:
good point. i've started thinking along these lines too, so more to come